share/settings/remote.go

```diff
@@ -46,7 +46,7 @@ func DecodeRemote(s string) (*Remote, error) {
 		s = strings.TrimPrefix(s, revPrefix)
 		reverse = true
 	}
-	parts := strings.Split(s, ":")
+	parts := regexp.MustCompile(`([A-Za-z0-9./]+|\[[A-Za-z0-9:/]+\]):?`).FindAllStringSubmatch(s, -1)
```

Comment:
Is `[A-Za-z0-9./]+` all valid characters for a hostname?

Maybe put the `[hex and colons]` match on the left hand side of the OR `|` and then make the hostname matcher more lax?

Reply:
That's a good point, this misses at least UTF8 domains and domains with hyphens.